operator/v1alpha1/types_image_content_source_policy.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageContentSource holds cluster-wide information about how to handle registry mirror rules.  The canonical name is `cluster`
+type ImageContentSource struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec ImageContentSourceSpec `json:"spec"`
+}
+
+type ImageContentSourceSpec struct {
+	// MirrorReposForDigests holds the registry/repo mirroring rules that are written to
+	// /etc/containers/registries.conf to be used by the runtime to determine which
+	// repo to fall back on, if the original repo cannot be accessed.
+	// We only support pulling by digests for mirrors.
+	// Will also be used in cases such as offline installs/upgrades.
+	// +optional
+	MirrorReposForDigests MirrorReposForDigests `json:"mirrorReposForDigests"`
```

Comment:
We don't use short names, and the order is mixed up.  In a declarative sense, you are declaring that you want any of these repositories to be substituted / used if an image is used in a pod with a digest and the repositories match

`repositoryDigestMirrors` is probably most accurate.  Remember to use the JSON field name in godoc, not the struct name.

This field should be a pointer since it references a struct that may be optional.

Reply:
Fixed up the name. I don't see why this needs to be a pointer though, other CRDs have optional fields that reference structs but are not pointers.